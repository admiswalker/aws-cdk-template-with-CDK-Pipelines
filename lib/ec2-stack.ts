import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as ec2 from 'aws-cdk-lib/aws-ec2';
import * as iam from 'aws-cdk-lib/aws-iam';
import * as fs from 'fs';


interface Ec2StackProps extends StackProps {
  prj_name: string;
  vpc: ec2.Vpc;
  ssm_iam_role: iam.Role;
}
export class Ec2Stack extends Stack {
  constructor(scope: Construct, id: string, props: Ec2StackProps) {
    super(scope, id, props);

    const cloud_config = ec2.UserData.forLinux({shebang: ''})
    const user_data_script = fs.readFileSync('./lib/ec2-stack_user-data.yaml', 'utf8');
    cloud_config.addCommands(user_data_script)
    const multipartUserData = new ec2.MultipartUserData();
    multipartUserData.addPart(ec2.MultipartBody.fromUserData(cloud_config, 'text/cloud-config; charset="utf8"'));
    
    const ec2_instance = new ec2.Instance(this, props.prj_name+'-'+this.constructor.name+'-general_purpose_ec2', {
      instanceType: new ec2.InstanceType('t3.nano'), // 1 Core, 1 GB
//    machineImage: ec2.MachineImage.genericLinux({'us-west-2': 'ami-XXXXXXXXXXXXXXXXX'}),
      machineImage: new ec2.AmazonLinuxImage({
        generation: ec2.AmazonLinuxGeneration.AMAZON_LINUX,
        edition: ec2.AmazonLinuxEdition.STANDARD,
        virtualization: ec2.AmazonLinuxVirt.HVM,
        storage: ec2.AmazonLinuxStorage.GENERAL_PURPOSE,
      }),
      vpc: props.vpc,
//    blockDevices: [{
//	    deviceName: '/dev/sda1',
//	    volume: ec2.BlockDeviceVolume.ebs(30),
//    }],
      vpcSubnets: props.vpc.selectSubnets({
        subnetGroupName: 'Private',
      }),
      role: props.ssm_iam_role,
      userData: multipartUserData,
    });

  }
}
