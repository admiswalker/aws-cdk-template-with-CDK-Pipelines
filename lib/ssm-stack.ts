import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as ec2 from 'aws-cdk-lib/aws-ec2';
import * as iam from 'aws-cdk-lib/aws-iam';


interface SsmStackProps extends StackProps {
  prj_name: string;
  vpc: ec2.Vpc;
}
export class SsmStack extends Stack {
  public readonly iam_role: iam.Role;

  constructor(scope: Construct, id: string, props: SsmStackProps) {
    super(scope, id, props);

    this.iam_role = new iam.Role(this, props.prj_name+'-'+this.constructor.name+'-iam_role_for_ssm', {
      assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
      managedPolicies: [
        iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMManagedInstanceCore'),
        iam.ManagedPolicy.fromAwsManagedPolicyName('CloudWatchAgentAdminPolicy'),
      ],
    });

    props.vpc.addInterfaceEndpoint(props.prj_name+'-'+this.constructor.name+'-InterfaceEndpoint_ssm', {
      service: ec2.InterfaceVpcEndpointAwsService.SSM,
    });
    props.vpc.addInterfaceEndpoint(props.prj_name+'-'+this.constructor.name+'-InterfaceEndpoint_ec2_messages', {
      service: ec2.InterfaceVpcEndpointAwsService.EC2_MESSAGES,
    });
    props.vpc.addInterfaceEndpoint(props.prj_name+'-'+this.constructor.name+'-InterfaceEndpoint_ssm_messages', {
      service: ec2.InterfaceVpcEndpointAwsService.SSM_MESSAGES,
    });

  }
}
