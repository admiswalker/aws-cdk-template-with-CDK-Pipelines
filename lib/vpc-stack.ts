import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as ec2 from 'aws-cdk-lib/aws-ec2';
import { InstanceType, NatInstanceImage, NatProvider } from 'aws-cdk-lib/aws-ec2';


interface VpcStackProps extends StackProps {
    prj_name: string;
}
export class VpcStack extends Stack {
  public readonly vpc: ec2.Vpc;

  constructor(scope: Construct, id: string, props: VpcStackProps) {
    super(scope, id, props);

    this.vpc = new ec2.Vpc(this, props.prj_name+'-'+this.constructor.name+'-vpc_for_ec2_and_ssm', {
      cidr: '10.0.0.0/16',
      natGateways: 1,
      natGatewayProvider: ec2.NatProvider.instance({
        instanceType: new InstanceType('t3.nano'),
        machineImage: new NatInstanceImage(),
      }),
      subnetConfiguration: [
        {
          name: 'Public',
          subnetType: ec2.SubnetType.PUBLIC,
          cidrMask: 27,
        },
        {
          name: 'Private',
          subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS,
          cidrMask: 27,
        },
      ],
    });

  }
}