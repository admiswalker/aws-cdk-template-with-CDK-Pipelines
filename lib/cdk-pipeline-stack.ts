import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as codecommit from 'aws-cdk-lib/aws-codecommit';
import {
    CodeBuildStep,
    CodePipeline,
    CodePipelineSource,
    ShellStep,
} from "aws-cdk-lib/pipelines";
import { CdkPipelineStageStack } from './cdk-pipeline-stage-stack';


interface CdkPipelineStackProps extends StackProps {
    prj_name: string;
}
export class CdkPipelineStack extends Stack {
  constructor(scope: Construct, id: string, props: CdkPipelineStackProps) {
    super(scope, id, props);

    // const repo = new codecommit.Repository(this, 'aws-cdk-template-with-CDK-Pipelines', { // Creates a new, empty repository
    //     repositoryName: 'aws-cdk-template-with-CDK-Pipelines',
    // });
    const repo = codecommit.Repository.fromRepositoryName(this, 'repo', 'aws-cdk-template-with-CDK-Pipelines'); // Use existing repository

    const pipeline = new CodePipeline(this, props.prj_name+'-'+this.constructor.name+'-aws-cdk-template-with-CDK-Pipelines',{
        pipelineName: props.prj_name+'-'+this.constructor.name+'-aws-cdk-template-with-CDK-Pipelines',
        synth: new CodeBuildStep('SynthStep', {
            input: CodePipelineSource.codeCommit(repo, 'main'),
            installCommands: [
                'npm install -g aws-cdk'
            ],
            commands: [
                'npm ci',
                'npm run build',
                'npx cdk synth'
            ],
        }),
    });

    const deploy_stack = new CdkPipelineStageStack(this, props.prj_name+'Deploy', {
        prj_name: props.prj_name,
        env: props.env,
    });
    const deployStage = pipeline.addStage(deploy_stack);
  }
}


// memo
// - AWS の CDK pipelines と aws-codepipeline の違い
//   https://zenn.dev/siroken3/scraps/769002c03b31d6
// - CDK Pipelineを導入するときの手順メモ
//   https://zenn.dev/pirosikick/scraps/a3d843cc54fcbf
// - class CodeCommitSourceAction
//   https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-codepipeline-actions.CodeCommitSourceAction.html
// - CREATE NEW PIPELINE - TypeScript Workshop
//   https://cdkworkshop.com/20-typescript/70-advanced-topics/200-pipelines/3000-new-pipeline.html
