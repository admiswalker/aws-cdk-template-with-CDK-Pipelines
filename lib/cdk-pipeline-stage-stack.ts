import { Construct } from 'constructs';
import { Stage, StageProps } from 'aws-cdk-lib';
import { AwsCdkTemplateStack } from '../lib/aws-cdk-template-stack';
import { VpcStack } from '../lib/vpc-stack';
import { SsmStack } from '../lib/ssm-stack';
import { Ec2Stack } from '../lib/ec2-stack';


interface CdkPipelineStageStackProps extends StageProps {
    prj_name: string;
}
export class CdkPipelineStageStack extends Stage {
  constructor(scope: Construct, id: string, props: CdkPipelineStageStackProps) {
    super(scope, id, props);

    const aws_cdk_template_stack = new AwsCdkTemplateStack(this, 'AwsCdkTemplateStack', {
        env: props.env,
      });
      
      const vpc_stack = new VpcStack(this, props.prj_name+'-VpcStack', {
        prj_name: props.prj_name,
        env: props.env,
      });
      
      const ssm_stack = new SsmStack(this, props.prj_name+'-SsmStack', {
        prj_name: props.prj_name,
        env: props.env,
        vpc: vpc_stack.vpc,
      });
      
      const ec2_stack = new Ec2Stack(this, props.prj_name+'-Ec2Stack', {
        prj_name: props.prj_name,
        env: props.env,
        vpc: vpc_stack.vpc,
        ssm_iam_role: ssm_stack.iam_role,
      });

  }
}

// memo
// - ADD APPLICATION TO PIPELINE - TypeScript Workshop
//   https://cdkworkshop.com/20-typescript/70-advanced-topics/200-pipelines/4000-build-stage.html
