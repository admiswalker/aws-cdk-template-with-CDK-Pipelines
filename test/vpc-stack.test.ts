import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { VpcStack } from '../lib/vpc-stack';


test('VpcStack', () => {
  
  const app = new cdk.App();

  const prj_name = 'AwsCdkTemplate';
  const env = {
    account: 'test-account',
    region: 'test-region'
  }

  // WHEN
  const stack = new VpcStack(app, prj_name+'-VpcStack', {
    prj_name: prj_name,
    env: env,
  });

  // THEN
  expect(Template.fromStack(stack).toJSON()).toMatchSnapshot();

});
