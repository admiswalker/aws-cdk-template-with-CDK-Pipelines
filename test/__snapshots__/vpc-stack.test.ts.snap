// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VpcStack 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "AwsCdkTemplateVpcStackvpcforec2andssm1E664AD5": Object {
      "Properties": Object {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmIGW1C32C75A": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmNatRole8BE205F0": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "ec2.",
                      Object {
                        "Ref": "AWS::URLSuffix",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmNatSecurityGroup96933567": Object {
      "Properties": Object {
        "GroupDescription": "Security Group for NAT instances",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "SecurityGroupIngress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "from 0.0.0.0/0:ALL TRAFFIC",
            "IpProtocol": "-1",
          },
        ],
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm",
          },
        ],
        "VpcId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssm1E664AD5",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet1DefaultRoute2813CB11": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "InstanceId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet1NatInstance2CE914A7",
        },
        "RouteTableId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet1RouteTableB94B9342",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet1RouteTableAssociation8D5F0311": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet1RouteTableB94B9342",
        },
        "SubnetId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet1Subnet36719CEC",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet1RouteTableB94B9342": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm/PrivateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssm1E664AD5",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet1Subnet36719CEC": Object {
      "Properties": Object {
        "AvailabilityZone": "dummy1a",
        "CidrBlock": "10.0.0.96/27",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm/PrivateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssm1E664AD5",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet2DefaultRouteFED1DC55": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "InstanceId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet1NatInstance2CE914A7",
        },
        "RouteTableId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet2RouteTable4DCED998",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet2RouteTable4DCED998": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm/PrivateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssm1E664AD5",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet2RouteTableAssociationD147F51A": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet2RouteTable4DCED998",
        },
        "SubnetId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet2Subnet1F1308D9",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet2Subnet1F1308D9": Object {
      "Properties": Object {
        "AvailabilityZone": "dummy1b",
        "CidrBlock": "10.0.0.128/27",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm/PrivateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssm1E664AD5",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet3DefaultRoute33678D92": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "InstanceId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet1NatInstance2CE914A7",
        },
        "RouteTableId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet3RouteTableFAED56E5",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet3RouteTableAssociationCE9C341A": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet3RouteTableFAED56E5",
        },
        "SubnetId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet3Subnet9EA21295",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet3RouteTableFAED56E5": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm/PrivateSubnet3",
          },
        ],
        "VpcId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssm1E664AD5",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPrivateSubnet3Subnet9EA21295": Object {
      "Properties": Object {
        "AvailabilityZone": "dummy1c",
        "CidrBlock": "10.0.0.160/27",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm/PrivateSubnet3",
          },
        ],
        "VpcId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssm1E664AD5",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet1DefaultRoute9F1F9984": Object {
      "DependsOn": Array [
        "AwsCdkTemplateVpcStackvpcforec2andssmVPCGWDDC683D5",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmIGW1C32C75A",
        },
        "RouteTableId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet1RouteTable435F9BE6",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet1NatInstance2CE914A7": Object {
      "DependsOn": Array [
        "AwsCdkTemplateVpcStackvpcforec2andssmNatRole8BE205F0",
      ],
      "Properties": Object {
        "AvailabilityZone": "dummy1a",
        "IamInstanceProfile": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet1NatInstanceInstanceProfile6C026AD8",
        },
        "ImageId": "ami-1234",
        "InstanceType": "t3.nano",
        "SecurityGroupIds": Array [
          Object {
            "Fn::GetAtt": Array [
              "AwsCdkTemplateVpcStackvpcforec2andssmNatSecurityGroup96933567",
              "GroupId",
            ],
          },
        ],
        "SourceDestCheck": false,
        "SubnetId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet1SubnetC12A5FB4",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm/PublicSubnet1/NatInstance",
          },
        ],
        "UserData": Object {
          "Fn::Base64": "#!/bin/bash",
        },
      },
      "Type": "AWS::EC2::Instance",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet1NatInstanceInstanceProfile6C026AD8": Object {
      "Properties": Object {
        "Roles": Array [
          Object {
            "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmNatRole8BE205F0",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet1RouteTable435F9BE6": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssm1E664AD5",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet1RouteTableAssociation87151265": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet1RouteTable435F9BE6",
        },
        "SubnetId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet1SubnetC12A5FB4",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet1SubnetC12A5FB4": Object {
      "Properties": Object {
        "AvailabilityZone": "dummy1a",
        "CidrBlock": "10.0.0.0/27",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssm1E664AD5",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet2DefaultRoute4FB124A2": Object {
      "DependsOn": Array [
        "AwsCdkTemplateVpcStackvpcforec2andssmVPCGWDDC683D5",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmIGW1C32C75A",
        },
        "RouteTableId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet2RouteTable1D01B6F1",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet2RouteTable1D01B6F1": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm/PublicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssm1E664AD5",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet2RouteTableAssociation27CAB38B": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet2RouteTable1D01B6F1",
        },
        "SubnetId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet2Subnet4450B837",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet2Subnet4450B837": Object {
      "Properties": Object {
        "AvailabilityZone": "dummy1b",
        "CidrBlock": "10.0.0.32/27",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm/PublicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssm1E664AD5",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet3DefaultRoute1BE989B2": Object {
      "DependsOn": Array [
        "AwsCdkTemplateVpcStackvpcforec2andssmVPCGWDDC683D5",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmIGW1C32C75A",
        },
        "RouteTableId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet3RouteTable793F0C90",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet3RouteTable793F0C90": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm/PublicSubnet3",
          },
        ],
        "VpcId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssm1E664AD5",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet3RouteTableAssociationFAF1C81D": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet3RouteTable793F0C90",
        },
        "SubnetId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet3Subnet44B370DA",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmPublicSubnet3Subnet44B370DA": Object {
      "Properties": Object {
        "AvailabilityZone": "dummy1c",
        "CidrBlock": "10.0.0.64/27",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "AwsCdkTemplate-VpcStack/AwsCdkTemplate-VpcStack-vpc_for_ec2_and_ssm/PublicSubnet3",
          },
        ],
        "VpcId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssm1E664AD5",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "AwsCdkTemplateVpcStackvpcforec2andssmVPCGWDDC683D5": Object {
      "Properties": Object {
        "InternetGatewayId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssmIGW1C32C75A",
        },
        "VpcId": Object {
          "Ref": "AwsCdkTemplateVpcStackvpcforec2andssm1E664AD5",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
